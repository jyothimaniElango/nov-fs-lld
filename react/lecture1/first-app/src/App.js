import logo from './logo.svg';
import './App.css';
import {ComponentOne} from './ComponentOne';
import ListComp from './ListComp';
import Counter from './Counter';
import TodoApp from './TodoApp/TodoApp';

function App() {
  return (
    <div className='App'>
       {/* I am at app.
       <ComponentOne name="Saptak" age={10} /> 
       <ComponentOne name="Scaler" age={20} />
       <ComponentOne name="React" age={30} />

       <ListComp list={['apple', 'Banana', 'Orange']} />
       <div style={{marginTop: '100px'}}>
       <ListComp list={['creta', 'verna', 'honda city', 'virtus']} />
       </div> */}

       {/* <Counter initialCount={10}/>

       <Counter initialCount={15} />

       <Counter initialCount={0}/> */}

       <TodoApp />
       
    </div>
  );
}

export default App;
