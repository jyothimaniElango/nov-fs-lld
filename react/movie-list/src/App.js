import {BrowserRouter as Router, Routes, Route, Link} from 'react-router-dom'; 
import './App.css';
import HomePage from './pages/HomePage';
import MovieDetails from './pages/MovieDetails';

function ErrorElement() {
  return (
    <div>
      <h2>404, Not found!</h2>
      <Link to={"./"}>Go to Home</Link>
    </div>
  );
}

function App() {
  return (
    <div className="App">
      <Router>
        <Routes>
          <Route path='/' element={<HomePage/>}></Route>
          <Route path='/movies/:moviesId' element={<MovieDetails />}></Route>
          <Route path='*' element={<ErrorElement />}></Route>
        </Routes>
      </Router>
    </div>
  );
}

export default App;
