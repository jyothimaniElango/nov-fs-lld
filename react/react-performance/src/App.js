import logo from './logo.svg';
import './App.css';
import MemoExample from './components/MemoExample';
import CallBackExample from './components/CallBackExample';

function App() {
  return (
    <div className="App">
      {/* <MemoExample /> */}
      {/* <CallBackExample /> */}
      <div>
        <p>Hello, a test implementation</p>
        <div>Hello world!!!</div>
        <div>One more div</div>
      </div>
    </div>
  );
}

export default App;
