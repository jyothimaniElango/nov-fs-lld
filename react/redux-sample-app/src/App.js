import { Provider } from 'react-redux';
import './App.css';
import Counter from './components/Counter';
import SingleUserComponent from './components/SingleUserComponent';
import TodoComponent from './components/TodoComponent';
import UsersComponent from './components/UsersComponent';
import store from './redux/store';

function App() {
  return (
    <div className="App">
      <Provider store={store} >
        {/* <Counter /> */}
        {/* <TodoComponent /> */}
        {/* <UsersComponent /> */}
        <SingleUserComponent />
      </Provider>
    </div>
  );
}

export default App;
