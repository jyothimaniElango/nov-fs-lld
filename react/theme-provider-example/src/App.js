import logo from './logo.svg';
import './App.css';
import NavBar from './components/NavBar';
import ProductDetails from './components/ProductDetails';
import ThemeProvider from './contenxt-api/ThemeProvider';

function App() {
  return (
    <div>
      <div className="App">
        <ThemeProvider>
          <NavBar /> 
          <ProductDetails />
        </ThemeProvider>
      </div>

      <div>
        <div>asdsad</div>
        <div></div>
      </div>
    </div>
  );
}

export default App;
