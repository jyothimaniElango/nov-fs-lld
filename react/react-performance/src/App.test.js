import { render, screen } from '@testing-library/react';
import App from './App';
import renderer from "react-test-renderer";

// test('renders learn react link', () => {
//   render(<App />);
//   const linkElement = screen.getByText(/learn react/i);
//   expect(linkElement).toBeInTheDocument();
// });

test('renders my main string properly', () => { 
  render(<App />);
  const pElement = screen.getByText(/Hello, a test implementation/i);
  expect(pElement).toBeInTheDocument();
});


test('renders hello world', () => { 
  render(<App />);
  const dElement = screen.getByText(/Hello world!!!/i);
  expect(dElement).toBeInTheDocument();
});

describe("Jest Snapshot testing suite", () => {
  it("Matches DOM Snapshot", () => {
    const domTree = renderer.create(<App />).toJSON();
    expect(domTree).toMatchSnapshot();
  });
});
