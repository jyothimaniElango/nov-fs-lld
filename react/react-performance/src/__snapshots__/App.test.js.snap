// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Jest Snapshot testing suite Matches DOM Snapshot 1`] = `
<div
  className="App"
>
  <div>
    <p>
      Hello, a test implementation
    </p>
    <div>
      Hello world!!!
    </div>
    <div>
      One more div
    </div>
  </div>
</div>
`;
